### ISSUE TITLE
clock not working

### ISSUE THREAD

**Issue by fvnky07** [ivan-lednev/obsidian-day-planner#649]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/ivan-lednev/obsidian-day-planner/issues?q=is%3Aissue+is%3Aopen) didn't find any similar reports.

### Expected Behavior

Clock should start and show up in calendar but nothing shows up, despite the metadata showing up below it.

https://github.com/user-attachments/assets/663b687a-0765-4bdf-aed0-20600a75405f

### Current behaviour

clock does not start and when i try to stop a "started" clock, it throws an error in the console.

### Steps to reproduce

tested on a fresh vault with no plugins except dataview and day planner and the issue persists.

### Which Operating Systems are you using?

- [ ] Android
- [ ] iPhone/iPad
- [X] Linux
- [ ] macOS
- [ ] Windows

### Obsidian Version

latest

**Comment by mkos** [comment]
Although I can't play the video, I think I have the same issue: Clock in and Clock out create appropriate entries below task, but it isn't reflected on timeline. Also when Clocked in, no active clocks are shown

<img width="624" alt="image" src="https://github.com/user-attachments/assets/07582f22-33fd-4959-901b-6d565bf9b380" />
<img width="528" alt="image" src="https://github.com/user-attachments/assets/78844097-fcb4-44fd-88b7-17296e979269" />

Im on MacOS Sequoia.

**Comment by fvnky07** [comment]
heres the video resent, the video is not working for me either.

https://github.com/user-attachments/assets/b5064e2b-95fa-43d6-b11c-e68539814e35

**Comment by byteZorvin** [comment]
Facing the same issue

**Comment by fvnky07** [comment]
after some experimenting, i found that setting the sources fixes the issue. click then options on the day planner and then specify the tag, for example, of the notes you are tracking. 
![image](https://github.com/user-attachments/assets/a61132c1-5c26-4ef1-86d9-0982c13577b7)

**Comment by fvnky07** [comment]
i believe that this isnt the intended functionality. i think the author intended non "filtered" notes to have time tracking. perhaps this is a feature not implemented, a limitation of dataview etc.

 I think this is still a bug so i wont close the issue.

**Comment by wrgrant** [comment]
Agreed this is a bug.

I did implement the filtering as shown in the screenshots from the README and found that the clocks work now.

In my case, my journal notes are in `VAULT/journal/daily` so I added `"journal/daily"` as the filter and it works now.

**Comment by mkos** [comment]
Yeah, I did like @wrgrant have done (i.e. used `"journal/daily"` as a filter, but the downside of it is that now tasks in the *entire* note are shown, not only those in the section `Day planner`. So if I have a section `Tomorrow` and tasks that later will be moved to the next note, they will appear in the timeline nevertheless and make it harder to plan the day.

**Comment by Liyuchuan** [comment]
The same problem，not work.

**Comment by hugovalepereira** [comment]
I am experiencing the same problem. I also found that applying a filter resolves the issue. However, it doesn't seem to matter whether the tag is related to any of the tasks being clocked. Any tag present anywhere in the daily note causes the Active Clocks to display as expected.

Examples:

Filtering for `#test` — Both clocks appear:
<img width="1792" alt="Image" src="https://github.com/user-attachments/assets/255e782a-19d5-4d99-970c-0993888f8a4a" />

Filtering for `#other` — Both clocks appear:
<img width="1792" alt="Image" src="https://github.com/user-attachments/assets/31ef8b96-0796-4207-a386-121c754a5244" />

Without any filters — Nothing shows:
<img width="1792" alt="Image" src="https://github.com/user-attachments/assets/01c616e9-37ae-4c45-9667-2e67c3d28c09" />

**Comment by bryanmaina** [comment]
I can confirm that this is the exact behavior that I am also getting.

https://github.com/ivan-lednev/obsidian-day-planner/issues/649#issuecomment-2633810731

**Comment by controlledentropy** [comment]
same here, so I'm sure that it's a BUG.
if you fills the "dataview source", the "Planner Heading Text" in settings will not affect so all the task will be load. 
else leave it in blank, planner will only load the tasks under 'Planner Heading Text', but time tracking and active clock will not work...

**Comment by kinglion580** [comment]
I have the same issue.

**Comment by ayrtonoropeza** [comment]
Same issue, still no fix. Hoping somebody can look into it soon.

**Comment by GustavoJMarynowski** [comment]
same issue

**Comment by linkion** [comment]
I was able to workaround this by adding a filter of `-"daily-notes/"` (my daily notes are stored with `/daily-notes/YYYY/MM-MMMM/YYYY-MM-DD-dddd` file structure, and the filter pulled all of it)

**Comment by ivan-lednev** [comment]
Fixed in 0.28.0. Please update and re-check.

**Comment by mkos** [comment]
It works now without filters. Thanks!
